jscomp/ml/ast_untagged_variants.ml

```diff
@@ -139,7 +169,10 @@ let get_block_type ~env (cstr : Types.constructor_declaration) :
     Some FloatType
   | true, Cstr_tuple [{desc = Tconstr (path, _, _)}]
     when Path.same path Predef.path_array ->
-    Some ArrayType
+    Some (InstanceType Array)
+  | true, Cstr_tuple [{desc = Tconstr (path, _, _)}]
+    when Path.same path Predef.path_promise ->
+    Some (InstanceType Promise)
```

Comment:
should promises and arrays be moved to `type_to_instanceof_backed_obj`?

Reply:
Yes, good idea.